addon/helpers/hook.js

```diff
@@ -6,10 +6,38 @@ import returnWhenTesting from 'ember-hook/utils/return-when-testing';
 
 const { Helper } = Ember;
 
-export default Helper.extend({
-  compute(params, qualifiers = {}) {
-    const [hook] = params;
+/**
+ * Test if an object is empty `Ember.isEmpty` surprisingly doesn't do that
+ * @see {@link https://github.com/emberjs/ember.js/issues/5543}
+ * @see {@link https://github.com/emberjs/ember.js/blob/v2.10.0/packages/ember-metal/lib/is_empty.js}
+ *
+ * @param {Object} obj - the object to test
+ * @returns {Boolean} true if there are no keys on the object
+ */
+function isEmpty (obj) {
```

Comment:
I was assuming the exception thrown by `Object.keys` would be sufficient, and that anything else for which `Object.keys` worked should be considered an object and treated as such, but if you'd like an explicit check, I can add that.